src/components/ContentWidgets/ToDoList/ToDoListEditor.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FormEvent, ChangeEvent } from 'react';
+import { WidgetEditorProps } from '../../ContentMapping/ContentMapping';
+import './ToDoList.css';
+
+export const ToDoListEditor: React.FC<WidgetEditorProps> = ({
+    originalContent, 
+    editedContent,
+    setEditedContentOnChange
+}) => {
+    let list = editedContent.toDoList || [];
+    let task_text = "";
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        task_text = e.target.value;
+    }
+
+    const addToDo = (e: FormEvent) => {
+        e.preventDefault();
+        let newList = [...list];
+        newList.push({index: newList.length, task: task_text});
+        setEditedContentOnChange("toDoList", newList);
+    }
+
+    const removeToDo = (indexToRemove: number) => {
+        let newList = [...list];
+        newList.splice(indexToRemove, 1);
+        for (let i = indexToRemove; i < newList.length; i++) {
```

Comment:
Probably not the best way to keep a unique order of these todo list items. I commented something similar on Elisa's, if you were to maintain an order you should probably have a `toDoList_order` and `toDoList_listItems` and use a hash, similar to how my widgets ordering is set up.

Reply:
Also try to stick to using `Array.forEach`. There are only very specific cases where you ever need a for loop like this which involve exiting the for loop early. In this case you don't exit early.